presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduSplit.java

```diff
@@ -29,15 +29,18 @@
     private final KuduTableHandle tableHandle;
     private final int primaryKeyColumnCount;
     private final byte[] serializedScanToken;
+    private final int bucketNumber;
 
     @JsonCreator
     public KuduSplit(@JsonProperty("tableHandle") KuduTableHandle tableHandle,
             @JsonProperty("primaryKeyColumnCount") int primaryKeyColumnCount,
-            @JsonProperty("serializedScanToken") byte[] serializedScanToken)
+            @JsonProperty("serializedScanToken") byte[] serializedScanToken,
+                     @JsonProperty("bucketNumber") int bucketNumber)
     {
         this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
         this.primaryKeyColumnCount = primaryKeyColumnCount;
         this.serializedScanToken = requireNonNull(serializedScanToken, "serializedScanToken is null");
+        this.bucketNumber = bucketNumber;
```

Comment:
check if the number is not negative ?

Reply:
I agree with you! The bucketNumber is created in a fori loop, so It will not be negative right now, but if some day people create a KuduSplit using this constructor, it will be necessary to do the check!